### PR TITLE
Add .mailmap file and us it in scripts/generate-AUTHORS.py.

### PR DESCRIPTION
**Pull request by mkrautz**
Also, re-generate the AUTHORS file as a self-test.

### CODE REVIEW

File: scripts/generate-AUTHORS.py [hunk 1]
```diff
@@ -232,14 +136,6 @@ def main():
 		if line == "zapman <unknown>":
```
Comment (hacst) [inline scripts/generate-AUTHORS.py]: What makes this one require special handling here?

File: scripts/generate-AUTHORS.py [hunk 2]
```diff
@@ -23,104 +23,8 @@
 	# Bots
```
Comment (hacst) [inline scripts/generate-AUTHORS.py]: Not directly part of this patch but in L21 ```"unknown <d0t@.(none)>",``` is another git configuration screwup by me and should probably be aliased accordingly.

**Comment by hacst** [comment]
Pretty cool. LGTM. See comments.

File: scripts/generate-AUTHORS.py (hunk 1 above)
Comment (mkrautz) [inline scripts/generate-AUTHORS.py]: We have no email for the guy. Git shows <unknown>, I like to show nothing at all in that case. That's pretty much it. I could make it more general, I suppose...

**Comment by mkrautz** [comment]
Addressed comments and landed as b03a6890f3c982544a6885971e72f3d8c94e8418.

**Pull request closed without merging**